test/region.c

```diff
@@ -201,9 +374,17 @@ int main()
 	region_basic();
 	region_test_truncate();
 	region_test_callbacks();
+	region_test_join();
+	region_test_alignment();
+#ifndef ENABLE_ASAN
 #ifndef NDEBUG
 	region_test_poison();
 #endif
+#else
+	region_test_poison();
+	region_test_tiny_reserve_size();
+	region_test_zero_alloc_truncation();
```

Comment:
The last two tests look like they would make sense for the default implementation. Can you rework them to enable them unconditionally?

Reply:
`region_test_zero_alloc_truncation` is deleted as now zero allocations are banned. `region_test_tiny_reserve_size` is still keept only for ASAN as discussed f2f (non-ASAN will not pass the test. It is slightly different in this respect.)